vernac/vernacentries.ml

```diff
@@ -2262,6 +2262,12 @@ let check_vernac_supports_polymorphism c p =
     | VernacExtend _ | VernacUniverse _ | VernacConstraint _) -> ()
   | Some _, _ -> user_err Pp.(str "This command does not support Polymorphism")
 
+(* Vernaculars that may handle a deprecation flag *)
+let check_vernac_supports_deprecation c d =
+  if d <> None then match c with
+  | VernacExtend _ -> ()
+  | _ -> user_err (Pp.str "This command does not support deprecation")
+
```

Comment:
I am sorry but I fail to see what the logic is here.

What you really want IMO anyways is a table that maps commands to supported features, and it is built from the declaration of the vernac.

Reply:
You are perfectly right. We need to design a good way for defining what attributes are supported and how they are handled.

My reasoning was that since the issue #8382 had a high-priority label, it would deserve this kind of quick-and-dirty patch mimicking what is done for other attributes. I agree that it’s quite bad to ignore attributes without notice.

Reply:
I mean, this check will forbid `#[deprecated] Definition`, right?

Reply:
Yes, it does. Is there already a mechanism for deprecating definitions?